sway-core/src/semantic_analysis/ast_node/expression/func_app_instantiation.rs

```diff
@@ -0,0 +1,121 @@
+use crate::build_config::BuildConfig;
+use crate::control_flow_analysis::ControlFlowGraph;
+use crate::error::*;
+use crate::semantic_analysis::{ast_node::*, TCOpts, TypeCheckArguments};
+use crate::type_engine::TypeId;
+use std::cmp::Ordering;
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn instantiate_function_application(
```

Comment:
should this be called `instantiate_` or should it be called `type_check_`? Because this seems more consistent with a type checking function, and perhaps should use `TypeCheckArguments`

Reply:
I modeled this function after `instantiate_enum` here: https://github.com/FuelLabs/sway/blob/fd6a093d50b8148f6dcd0e75dfbf8d39d43102d0/sway-core/src/semantic_analysis/ast_node/expression/enum_instantiation.rs#L10